main.tf

```diff
@@ -1,16 +1,24 @@
+data "azurerm_resource_group" "rg" {
+  count = var.create_resource_group ? 0 : 1
+  name  = var.resource_group_name
+}
+
 # Create rgs as defined by var.virtual_wan
 resource "azurerm_resource_group" "rg" {
-  for_each = var.resource_group_name != null ? toset([var.resource_group_name]) : toset([])
-
+  count    = var.create_resource_group ? 1 : 0
   location = var.location
   name     = var.resource_group_name
   tags     = try(merge(var.resource_group_tags, var.tags), {})
 }
 
+locals {
+  resource_group_name = var.create_resource_group ? azurerm_resource_group.rg[0].name : data.azurerm_resource_group.rg[0].name
```

Comment:
By doing a data source lookup would the data source not execute potentially before the resource group is created or if there is a lifecycle recreation of the resource group the lookup could occur while the resource exists but would not necessarily be valid when it goes to be utilized?

When I solved this in my fork to make the module functional I treated the resource group that wasn't created inside the module as simply a variable input and passed it through appropriately rather than getting into trying to manage the lifecycle of the resource outside my scope inside the module.

Reply:
Hi. The resource and data source are mutually exclusive. The data source will only be used if you supply a pre-existing resource group. The resource will only be used if you create a new resource group.

Reply:
See the `count` condition on each.